dist/common/scripts/scylla_util.py

```diff
@@ -351,6 +351,199 @@ def user_data(self):
         except urllib.error.HTTPError:  # empty user-data
             return ""
 
+class azure_instance:
+    """Describe several aspects of the current Azure instance"""
+
+    EPHEMERAL = "ephemeral"
+    ROOT = "root"
+    GETTING_STARTED_URL = "http://www.scylladb.com/doc/getting-started-azure/"
+    META_DATA_BASE_URL = "http://169.254.169.254/1.0/meta-data/instance/"
+    ENDPOINT_SNITCH = "GossipingPropertyFileSnitch"
+
+    def __init__(self):
+        self.__type = None
+        self.__cpu = None
+        self.__location = None
+        self.__zone = None
+        self.__memoryGB = None
+        self.__nvmeDiskCount = None
+        self.__firstNvmeSize = None
+        self.__osDisks = None
+
+    @staticmethod
+    def is_azure_instance():
+        """Check if it's Azure instance via DNS lookup to metadata server."""
+        try:
+            addrlist = socket.getaddrinfo('metadata.azure.internal', 80)
+        except socket.gaierror:
+            return False
+        for res in addrlist:
+            af, socktype, proto, canonname, sa = res
+            if af == socket.AF_INET:
+                addr, port = sa
+                if addr == "169.254.169.254":
+                    return True
+        return False
+
+# as per https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service?tabs=windows#supported-api-versions
+    API_VERSION="?api-version=2021-01-01"
+
+    def __instance_metadata(self, path):
+        """query Azure metadata server"""
+        return curl( self.META_DATA_BASE_URL + path + self.API_VERSION +"&format=text")
+
+    def _non_root_nvmes(self):
+        """get list of nvme disks from os, filter away if one of them is root"""
+        nvme_re = re.compile(r"nvme\d+n\d+$")
+
+        root_dev_candidates = [x for x in psutil.disk_partitions() if x.mountpoint == "/"]
+        if len(root_dev_candidates) != 1:
+            raise Exception("found more than one disk mounted at root ".format(root_dev_candidates))
+
+        root_dev = root_dev_candidates[0].device
+        # if root_dev.startswith("/dev/mapper"):
+        #     raise Exception("mapper used for root, not checking if nvme is used ".format(root_dev))
```

Comment:
What's this comment?

Reply:
removed, sorry,
I am actually tempted to replace all this method with https://github.com/scylladb/scylla/pull/8444
but it's not merged yet

Reply:
replaced it with method from 8444